components/script/dom/location.rs

```diff
@@ -242,7 +256,11 @@ impl LocationMethods for Location {
     // https://html.spec.whatwg.org/multipage/#dom-location-protocol
     fn SetProtocol(&self, value: USVString) -> ErrorResult {
         self.check_same_origin_domain()?;
-        self.set_url_component(value, UrlHelper::SetProtocol);
+        // Step 6: If copyURL's scheme is not an HTTP(S) scheme, then terminate these steps.
+        let scheme = value.split(':').next().unwrap();
```

Comment:
Isn’t it dangerous to have `unwrap()` here?

Reply:
Yes you're right, actually per Step 5 of the protocol setter spec, we should return a `Error::Syntax` here, see https://html.spec.whatwg.org/multipage/history.html#dom-location-protocol

Reply:
Would my latest commit fix that?

Reply:
Yes. 

Please run `./mach fmt` to fix https://community-tc.services.mozilla.com/tasks/VcHxks51RSywXhLH0W3dXg/runs/0/logs/https%3A%2F%2Fcommunity-tc.services.mozilla.com%2Fapi%2Fqueue%2Fv1%2Ftask%2FVcHxks51RSywXhLH0W3dXg%2Fruns%2F0%2Fartifacts%2Fpublic%2Flogs%2Flive.log

And please also squash...